### PR TITLE
Validate stix bundles before search

### PR DESCRIPTION
**Pull request by mdazam1942**
bundle_validator folder contains a script `bundle_validator.py` and a readme with instruction how to run the scrip to validate a bundle file

### CODE REVIEW

File: stix_shifter_modules/stix_bundle/configuration/config.json [hunk 1]
```diff
@@ -13,6 +13,16 @@
...
             "default": "https://www.ibm.com/support/knowledgecenter/SSTDPP_1.1.0/docs/scp-core/data-sources-stix.html"
+        },
+        "options": {
+            "type": "fields",
+            "stix_validator": {
+                "type": "boolean",
+                "default": true,
+                "optional": true,
+                "hidden": true,
+                "previous": "connection.stix_validator"
```
Comment (yurii-klymenko) [inline stix_shifter_modules/stix_bundle/configuration/config.json:24]: we have that value already, why did you insert it there: https://github.com/opencybersecurityalliance/stix-shifter/blob/develop/stix_shifter_modules/config.json#L45-L50

**Review by yurii-klymenko — CHANGES_REQUESTED** [review]

File: stix_shifter_modules/stix_bundle/configuration/config.json (hunk 1 above)
Comment (mdazam1942) [reply stix_shifter_modules/stix_bundle/configuration/config.json:24]: so I needed to set the default value `"default": true,`

base config.json doesn't have any default parameter.

Comment (mdazam1942) [reply stix_shifter_modules/stix_bundle/configuration/config.json:24]: > we have that value already, why did you insert it there: https://github.com/opencybersecurityalliance/stix-shifter/blob/develop/stix_shifter_modules/config.json#L45-L50

so I needed to set the default value "default": true,

base config.json doesn't have any default parameter.

Comment (yurii-klymenko) [reply stix_shifter_modules/stix_bundle/configuration/config.json:24]: it should be enough to specify:
`        "options": {
            "stix_validator": {
                "default": true
            }
        }`

Comment (yurii-klymenko) [reply stix_shifter_modules/stix_bundle/configuration/config.json:24]: and please set defalut:false in the base config.

Comment (mdazam1942) [reply stix_shifter_modules/stix_bundle/configuration/config.json]: @yurii-klymenko updated

**Comment by mdazam1942** [comment]
We may need to confirm with UDI about the bundle validation on ping stage.

File: stix_shifter_modules/stix_bundle/configuration/config.json [hunk 2]
```diff
@@ -13,6 +13,12 @@
         "help": {
             "type": "link",
             "default": "https://www.ibm.com/support/knowledgecenter/SSTDPP_1.1.0/docs/scp-core/data-sources-stix.html"
+        },
+        "options": {
+            "type": "fields",
```
Comment (yurii-klymenko) [inline stix_shifter_modules/stix_bundle/configuration/config.json:18]: no need to specify type:fields twice

Comment (mdazam1942) [reply stix_shifter_modules/stix_bundle/configuration/config.json]: fixed

File: bundle_validator/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Validate STIX bundle
+
+You can easily validate your stix bundle file by following the below steps:
+
+## Prerequisites:
+
+Make sure you have Python 3 installed and a shell terminal available in your machine. 
+
+If you have not already cloned the [stix-shifter github project](https://github.com/opencybersecurityalliance/stix-shifter), you need to install GIT in order to clone the STIX-shifter git repository:
```
Comment (delliott90) [inline bundle_validator/README.md:9]: You should just mention that you need git installed in the line above. Right now it looks like the following terminal command is to install git, but it's really to clone the repo.

File: bundle_validator/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+2. cd into /stix-shifter/bundle_validator
+3. Copy your STIX bundle JSON file into the bundle_validator directory
+4. Run the validate.sh script. You need to specify the bundle json file name:
+    ```
+    ./validate.sh <STIX Bundle JSON FIle>
+    ```
+5. After successfull validation, you should see a messegae in your terminal: `STIX Bundle validated!!`
+6. For unsuccessfull validation, you should see mainly two types of error-
+    1. If JSON format is invalid: `Malformed JSON in the STIX Bundle: <ERROR details>`
+    2. If the file contains invalid STIX Objects, you should see errors/warnings with heading `[X] STIX JSON: Invalid`. It is mandatory to fix the errors red marked as `[X]`. Warnings which are yellow marked as `[!]`, can be ingnored but recommended to fix. For example-
```
Comment (delliott90) [inline bundle_validator/README.md:30]: change to "marked red" and "marked yellow"

File: bundle_validator/README.md (hunk 3 above)
Comment (mdazam1942) [reply bundle_validator/README.md]: updated

File: bundle_validator/output.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
+{
```
Comment (delliott90) [inline bundle_validator/output.json:1]: What's the output file used for?

Comment (mdazam1942) [reply bundle_validator/output.json]: ops that's a test file. removed

File: bundle_validator/README.md (hunk 4 above)
Comment (mdazam1942) [reply bundle_validator/README.md]: done

Comment (delliott90) [reply bundle_validator/README.md]: missed changing to "marked yellow".

**Review by yurii-klymenko — APPROVED** [review]

File: bundle_validator/README.md (hunk 4 above)
Comment (mdazam1942) [reply bundle_validator/README.md]: I thought I changed it . 
```
2. If the file contains invalid STIX Objects, you should see errors/warnings with heading `[X] STIX JSON: Invalid`. It is mandatory to fix the errors marked red as `[X]`. Warnings which are marked yellow as `[!]`, can be ingnored but recommended to fix. For example-
```

**Review by delliott90 — APPROVED** [review]

**Pull request merged**